### DISCUSSION TITLE
Client: OPTIONS cannot be sent when in playing state

### DISCUSSION THREAD

**Discussion by tmatth** [bluenviron/gortsplib#70] [Questions]
If sending an OPTIONS request while the state is currently playing, it fails with:
> panic: must be in state [initial prePlay preRecord], while is in state play

This seems to contradict both https://datatracker.ietf.org/doc/html/rfc7826#section-13.1 and https://datatracker.ietf.org/doc/html/rfc2326#section-10.1 which both state:
> An OPTIONS request may be issued at any time

**Comment by aler9** [comment]
Hello, when a client is reading or publishing, OPTIONS requests are sent to the server indeed, but the procedure is entirely managed by the library, since:

* OPTIONS requests are used as keep-alives
* when reading or publishing, client is already reading data from the server, therefore it can't "simply" read a response, since there's already a routing that is reading

Why do you need to send OPTIONS requests manually?

**Comment by tmatth** [comment]
> Hello, when a client is reading or publishing, OPTIONS requests are sent to the server indeed, but the procedure is entirely managed by the library, since:
> 
>     * OPTIONS requests are used as keep-alives
> 
>     * when reading or publishing, client is already reading data from the server, therefore it can't "simply" read a response, since there's already a routing that is reading
> 
> 
> Why do you need to send OPTIONS requests manually?

It was for testing shorter keep alive intervals and GET_PARAMETER vs. OPTIONS as keepalives, I think it's safe to close this issue since the underlying problem I was debugging turned out to be downstream of rtsp-simple-server/gortsplib, sorry for the noise.
